src/game/savegame.c

```diff
@@ -433,13 +434,18 @@ bool Savegame_UpdateDeathCounters(int32_t slot_num, GAME_INFO *game_info)
 
 void Savegame_Shutdown(void)
 {
-    for (int i = 0; i < MAX_SAVE_SLOTS; i++) {
-        SAVEGAME_INFO *savegame_info = &m_SavegameInfo[i];
-        savegame_info->format = 0;
-        savegame_info->counter = -1;
-        savegame_info->level_num = -1;
-        Memory_FreePointer(&savegame_info->full_path);
-        Memory_FreePointer(&savegame_info->level_title);
+    if (m_SavegameInfo == NULL) {
```

Comment:
Shutdown functions are about cleaning things up, freeing memory and resetting back to neutral state. They should not allocate new memory.
Please prepare a new function `Savegame_Init(void)` that does the `Memory_Alloc` call, and in `Savegame_Shutdown`, free that memory.

Reply:
Agreed. Then I think you can just do:
```
void Savegame_ScanSavedGames(void)
{
    Savegame_Shutdown();
    Savegame_Init();
    ...
```

Reply:
Doing `Savegame_Init` after `Savegame_Shutdown` seg faults, the other way around works though.